File: csharp-solution/E-CommerceSystem/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace E_CommerceSystem.Models
{
  public class Product
  {
    public Product(string name, double price, int quantiy)
    {
      Name = name;
      Price = price;
      Quantiy = quantiy;
    }
  
    public string Name { get; protected set; }

    public double Price { get; protected set; }

    public int Quantiy { get; protected set; }

    public bool IsAvailable (int amount)
    {
      return Quantiy >= amount;        
    }
    public void ReduceQuantity(int amount)
    {
        Quantiy -= amount;
    }
    public virtual bool IsExpired()
    {
      return false;
    }

  }
}

File: java-solution/src/Model/Product.java
package Model;

import interfaces.IProduct;

public class Product implements IProduct {
    protected String name;
    protected double price;
    protected int quantity;
    public  Product (String name ,double price ,int quantity){
        this.name=name;
        this.price=price;
        this.quantity=quantity;
    }
    public String getName(){
        return name;
    }
    public double getPrice(){
        return price;
    }
   public  boolean isAvailable(int amount){
        return quantity>=amount;
   }
    public void reduceQuantity(int amount){
        quantity-=amount;
    }

}
